appinventor/components-ios/src/Sprite.swift

```diff
@@ -403,7 +403,53 @@ open class Sprite: ViewComponent, UIGestureRecognizerDelegate {
     updateY(y: y)
     registerChanges()
   }
+
+  /**
+   * Moves the sprite to given point after checking whether the coordinates are numbers.
+   */
+  @objc open func MoveToPoint(_ coordinates: YailList<AnyObject>) {
+    let xCoord = CoerceToDouble(coordinates[1] as AnyObject)
+    let yCoord = CoerceToDouble(coordinates[2] as AnyObject)
+    self.MoveTo(xCoord, yCoord)
+  }
+
+  /**
+   * Helper function for MoveToPoint
+   */
+  func CoerceToDouble(_ coordinate: AnyObject ) -> Double {
+    //unpack coordinate (check if are numbers and parse if string)
+    if let number = coordinate as? NSNumber {
+      return number.doubleValue
+    } else if let number = Double(coordinate as? NSString as? String ?? "") {
+      return number
+    } else {
+      return Double.nan
+    }
+  }
+
+  /**
+   * Moves the sprite to given point after checking whether the coordinates are numbers.
+   */
+  @objc open func MoveToPoint(_ coordinates: YailList<AnyObject>) {
+    let xCoord = coerceToDouble(coordinates[1] as AnyObject)
+    let yCoord = coerceToDouble(coordinates[2] as AnyObject)
+    self.MoveTo(xCoord, yCoord)
+  }
   
+  /**
+   * Helper function for MoveToPoint
+   */
+  private func coerceToDouble(_ coordinate: AnyObject ) -> Double {
+      //unpack coordinate (check if are numbers and parse if string)
+      if let number = coordinate as? NSNumber {
+        return number.doubleValue
+      } else if let number = Double(coordinate as? NSString as? String ?? "") {
+        return number
+      } else {
+        return Double.nan
+      }
+    }
+    
```

Comment:
Why did you duplicate the code here? The change I asked you to make previously should have only been a few lines maximum and not required duplicating the code.

Reply:
My bad! I think there might have been some confusion that caused repeated code before I did git stash pop. Just made a new commit that removed the duplicate code!